### PR TITLE
feat(Guidebanner): convert to .tsx

### PR DESCRIPTION
**Pull request by paul-balchin-ibm**
Contributes to #4487, #4488, #4489, #4490

This PR converts `Guidebanner`, `GuidebannerElement`, `GuidebannerElementButton`, and `GuidebannerElementLink`  to TypeScript.

#### What did you change?

- file extension
- TS interface

#### How did you test and verify your work?

storybook build and type checks

### CODE REVIEW

**Review by annawen1 — APPROVED** [review]
LGTM

File: packages/ibm-products/src/components/Guidebanner/GuidebannerElement.tsx [hunk 1]
```diff
@@ -16,29 +16,47 @@ import { pkg } from '../../settings';
...
+   */
+  button: ReactNode;
+  /**
+   * Provide an optional class to be applied to the containing node.
+   */
+  className?: string;
+  /**
+   * The description of the element.
+   */
+  description: string | ReactNode;
```
Comment (matthewgallo) [inline packages/ibm-products/src/components/Guidebanner/GuidebannerElement.tsx:32]: I believe `string` is also considered a `ReactNode` so you could just specify `description: ReactNode`
```suggestion
  description: ReactNode;
```

File: packages/ibm-products/src/components/Guidebanner/GuidebannerElementButton.tsx [hunk 2]
```diff
@@ -18,23 +18,58 @@ import { pkg } from '../../settings';
...
+  (
+    { children, className, type, ...rest }: GuidebannerElementButtonProps,
+    ref
+  ) => {
+    if (type === 'primary') {
+      return (
+        <Button
+          {...rest}
+          className={cx(blockClass, className)}
+          iconDescription={'Crossroads'}
```
Comment (matthewgallo) [inline packages/ibm-products/src/components/Guidebanner/GuidebannerElementButton.tsx:55]: Can you update this so it's not a hard coded string? This component might need a new prop, `iconDescription`

**Review by matthewgallo — CHANGES_REQUESTED** [review]
Just had two comments regarding `string` and `ReactNode` and a hardcoded string in `GuideBannerElementButton`

**Comment by paul-balchin-ibm** [comment]
Closing. Made changes, merged latest, push was rejected, _followed git instructions to fix_, now have other people's gazillion changes.

Please see replacement PR #5321

**Pull request closed without merging**